File: integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/book/BookShortView.java
package io.quarkus.it.mongodb.panache.book;

import java.time.LocalDate;

import javax.json.bind.annotation.JsonbDateFormat;

import io.quarkus.mongodb.panache.common.ProjectionFor;

@ProjectionFor(Book.class)
public class BookShortView {
    private String title; // uses the field name title and not the column name bookTitle
    private String author;
    @JsonbDateFormat("yyyy-MM-dd")
    private LocalDate creationDate;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public LocalDate getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(LocalDate creationDate) {
        this.creationDate = creationDate;
    }
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/book/BookShortView.kt
package io.quarkus.it.mongodb.panache.book

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonFormat.Shape
import io.quarkus.mongodb.panache.common.ProjectionFor
import java.time.LocalDate

@ProjectionFor(Book::class)
class BookShortView {
    // uses the field name title and not the column name bookTitle
    var title: String? = null
    var author: String? = null

    @JsonFormat(shape = Shape.STRING, pattern = "yyyy-MM-dd") var creationDate: LocalDate? = null
}
